### PR TITLE
Implemented urlrewriter for nnm-club.me

### PR DESCRIPTION
**Pull request by asm0dey**

### CODE REVIEW

File: flexget/plugins/urlrewrite_nnmclub.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class UrlRewriteNnmClub(object):
+    """Nnm-club.me urlrewriter."""
+
+    def url_rewritable(self, task, entry):
+        return entry['url'].startswith('http://nnm-club.me/forum/viewtopic.php?t=')
+
+    def url_rewrite(self, task, entry):
+        initUrl = entry['url']
+        html = urllib2.urlopen(initUrl).read()
```
Comment (cvium) [inline flexget/plugins/urlrewrite_nnmclub.py]: Generally, we use the `requests` library instead of urllib.

Comment (asm0dey) [inline flexget/plugins/urlrewrite_nnmclub.py]: fixed

File: flexget/plugins/urlrewrite_nnmclub.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+import requests
```
Comment (cvium) [inline flexget/plugins/urlrewrite_nnmclub.py]: We actually have a requests util, so it should be `from flexget.utils import requests` instead. Sorry :D

Comment (asm0dey) [inline flexget/plugins/urlrewrite_nnmclub.py]: Fixed that too

File: flexget/plugins/urlrewrite_nnmclub.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+from flexget.utils import requests
```
Comment (cvium) [inline flexget/plugins/urlrewrite_nnmclub.py]: One last thing lol. We generally have the imports in 3 categories. "Standard python libs", "other libs" and "flexget imports" :P

File: flexget/plugins/urlrewrite_nnmclub.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+class UrlRewriteNnmClub(object):
+    """Nnm-club.me urlrewriter."""
+
+    def url_rewritable(self, task, entry):
+        return entry['url'].startswith('http://nnm-club.me/forum/viewtopic.php?t=')
+
+    def url_rewrite(self, task, entry):
+        initUrl = entry['url']
+        html = requests.get(initUrl).content
```
Comment (gazpachoking) [inline flexget/plugins/urlrewrite_nnmclub.py]: We could actually just use `task.requests` here, that means plugins like `modify_headers` will work with this plugin's requests.

Comment (asm0dey) [inline flexget/plugins/urlrewrite_nnmclub.py]: What is task.requests?

Comment (cvium) [inline flexget/plugins/urlrewrite_nnmclub.py]: The task object that you take as input to url_rewrite contains a requests object that you can use directly. In other words: Change line 21 to `html = tasks.requests.get(initUrl).content`.

File: flexget/plugins/urlrewrite_nnmclub.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    def url_rewritable(self, task, entry):
+        return entry['url'].startswith('http://nnm-club.me/forum/viewtopic.php?t=')
+
+    def url_rewrite(self, task, entry):
+        html = task.requests.get(entry['url']).content
+        soup = BeautifulSoup(html)
+        links = soup.findAll('a')
+        hrefs = map(lambda x: x.get('href'), links)
+        entry['url'] = filter(lambda x: x != None and x.startswith('magnet'), hrefs)[0]
```
Comment (cvium) [inline flexget/plugins/urlrewrite_nnmclub.py]: Won't this crash unexpectedly if it doesn't find a magnet url?

**Pull request merged**